ppa/archive/views.py

```diff
@@ -166,3 +170,73 @@ def render_to_csv(self, data):
 
     def get(self, *args, **kwargs):
         return self.render_to_csv(self.get_data())
+
+
+class BulkAddCollectionView(ListView, FormMixin, ProcessFormView):
+    '''
+    View to bulk add a queryset of :class:ppa.archive.models.DigitizedWork
+    to a set of :class:ppa.archive.models.Collection instances.
+
+    Restricted to staff users via staff_member_required on url.
+    '''
+
+    model = DigitizedWork
+    template_name = 'admin/archive/digitizedwork/bulk_add_collections.html'
+    form_class = BulkAddCollectionForm
+    success_url = reverse_lazy('admin:archive_digitizedwork_changelist')
+
+    def get_queryset(self, *args, **kwargs):
+        '''Return a queryset filtered by id, or empty list if no ids'''
+        ids = self.request.GET.get('ids', None)
+        # if somehow a problematic non-pk is pushed, will be ignored in filter
+        return DigitizedWork.objects.filter(id__in=ids.split(',')
+                                            if ids else [])
+
+    def get_initial(self):
+        '''
+        Set the hidden field with the :class:ppa.archive.models.DigitizedWork
```

Comment:
I'm not clear what you're doing here; please clarify in the docstring and/or comments.  Is this just overriding the default method to filter out any invalid ids?

Reply:
I've attempted to clarify this with a new comment while refactoring.